File: archive/p/php/selection-sort.php
function SelectionSort(&$data, $count) {
	for ($i = 0; $i < $count - 1; $i++)
	{
		$min = $i;

		for ($j = $i + 1; $j < $count; $j++)
		{
			if ($data[$j] < $data[$min])
			{
				$min = $j;
			}
		}

		$temp = $data[$min];
		$data[$min] = $data[$i];
		$data[$i] = $temp;
	}
}

File: archive/m/matlab/selection_sort.m
function sorted = selection_sort(array)
  if nargin == 0 || length(array)<=1
      fprintf("Usage: please provide a list of at least two integers to sort in the format [1 2 3 4 5]\n");
      return
  end  
  
  sorted = [];
  while ~isempty(array)
      x = min(array);
      sorted = [ sorted array(array==x) ];
      array(array==x) = [];  
  end
end